apps/gdalalg_raster_proximity.cpp

```diff
@@ -0,0 +1,251 @@
+/******************************************************************************
+ *
+ * Project:  GDAL
+ * Purpose:  gdal "raster proximity" subcommand
+ * Author:   Alessandro Pasotti <elpaso at itopen dot it>
+ *
+ ******************************************************************************
+ * Copyright (c) 2025, Alessandro Pasotti <elpaso at itopen dot it>
+ *
+ * SPDX-License-Identifier: MIT
+ ****************************************************************************/
+
+#include "gdalalg_raster_proximity.h"
+
+#include "cpl_conv.h"
+#include "cpl_vsi_virtual.h"
+
+#include "gdal_alg.h"
+#include "gdal_priv.h"
+#include "gdal_utils.h"
+#include "commonutils.h"
+
+//! @cond Doxygen_Suppress
+
+#ifndef _
+#define _(x) (x)
+#endif
+
+/************************************************************************/
+/*      GDALRasterProximityAlgorithm::GDALRasterProximityAlgorithm()            */
+/************************************************************************/
+
+GDALRasterProximityAlgorithm::GDALRasterProximityAlgorithm()
+    : GDALAlgorithm(NAME, DESCRIPTION, HELP_URL)
```

Comment:
Would it make sense to inherit `GDALRasterPipelineStepAlgorithm` to reduce some I/O boilerplate ?

Reply:
That isn't possible. The proximity algorithm requires 2 passes and is not compatible of streaming/pipelining